src/components/token/tests/helper/create-id-token.test.ts

```diff
@@ -108,4 +112,110 @@ describe("createIdToken tests", () => {
     });
     expect(typeof tokenParts[2]).toBe("string");
   });
+
+  it("returns an invalid header if the client config has enabled INVALID_ALG_HEADER", async () => {
+    tokenSigningAlgorithmSpy.mockReturnValue("ES256");
+    subSpy.mockReturnValue(testSubClaim);
+    clientIdSpy.mockReturnValue(testClientId);
+    idTokenErrorSpy.mockReturnValue(["INVALID_ALG_HEADER"]);
+
+    const idToken = await createIdToken(mockAuthRequestParams, testAccessToken);
+    const tokenParts = idToken.split(".");
+
+    const header = decodeTokenPart(tokenParts[0]);
+
+    expect(tokenParts.length).toBe(3);
+    expect(header).toStrictEqual({
+      alg: "HS256",
+    });
+  });
+
+  it("returns an invalid signature if the client config has enabled INVALID_SIGNATURE enabled", async () => {
+    tokenSigningAlgorithmSpy.mockReturnValue("ES256");
+    subSpy.mockReturnValue(testSubClaim);
+    clientIdSpy.mockReturnValue(testClientId);
+    idTokenErrorSpy.mockReturnValue(["INVALID_SIGNATURE"]);
+
+    const idToken = await createIdToken(mockAuthRequestParams, testAccessToken);
+
+    const ecTokenKey = await importPKCS8(EC_PRIVATE_TOKEN_SIGNING_KEY, "ES256");
+
+    await expect(jwtVerify(idToken, ecTokenKey)).rejects.toThrow(
+      errors.JWSSignatureVerificationFailed
+    );
+  });
+
+  it("returns a token with iat in the future if the client config has enabled TOKEN_NOT_VALID_YET enabled", async () => {
+    tokenSigningAlgorithmSpy.mockReturnValue("ES256");
+    subSpy.mockReturnValue(testSubClaim);
+    clientIdSpy.mockReturnValue(testClientId);
+    idTokenErrorSpy.mockReturnValue(["TOKEN_NOT_VALID_YET"]);
+
+    const idToken = await createIdToken(mockAuthRequestParams, testAccessToken);
+
+    const payload = decodeTokenPart(idToken.split(".")[1]);
+    expect(payload.iat).toEqual(Math.floor(testTimestamp / 1000) + 86400);
```

Comment:
might make it a bit clearer to re-name _testTimestamp_ to _testTimestampMilliseconds_ and create a constant called _secondsInADay_ for the offset?

Reply:
Have renamed the timestamp to be `testTimestampMs` and created a constant `oneDayInSeconds` for the offset